clang/test/CodeGenSYCL/launch_bounds_nvptx.cpp

```diff
@@ -0,0 +1,105 @@
+// RUN: %clang_cc1 -fsycl-is-device -internal-isystem %S/Inputs -sycl-std=2017 -triple nvptx-unknown-unknown -target-cpu sm_90 -disable-llvm-passes -S -emit-llvm -o - %s | FileCheck %s
```

Comment:
Is the `-sycl-std=2017` neccesary here?

Reply:
It would appear so, without it https://github.com/intel/llvm/pull/11192/files/80cc27930e7e5ba5c78457fd59ae15b8c78dc81d#diff-ac42d36bc2581b49ce17b21a3e340962dad87a9f9dae569e239c2189ae6e2986R54 does not generate any attribute metadata.

Reply:
I followed what other tests in `CodeGenSYCL`, dealing with attributes were doing, for example: [required sub group size](https://github.com/intel/llvm/blob/9431687c6bfe57ab8f06cdabff4273a73c0abf9f/clang/test/CodeGenSYCL/reqd-sub-group-size.cpp). If you think it's more appropriate to drop the 2017 (and propagation tests that require it), I'm happy to do so?

Reply:
No I'm ok with this. I was just making sure that you were :)

Reply:
It really depends on he functionality you want. If you want the attribute propagated to the kernel in SYCL 1.2.1, this is ok. If you don't want it doing that or this functionally doesn't make sense, maybe remove it. I leave it up to you because I am not familiar with required functionality here.

Reply:
>>It really depends on he functionality you want. If you want the attribute propagated to the kernel in SYCL 1.2.1, this is ok. If you don't want it doing that or this functionally doesn't make sense, maybe remove it. 

Agreed

Reply:
It feels like keeping the same experience as other attributes have is beneficial here, going to leave it as it is.